lab-ron/chat/server.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+
+const net = require('net');
+const EE = require('events');
+const Client = require('../model/client.js');
+
+const PORT = process.env.PORT || 3000;
+const pool = [];
+const server = net.createServer();
+const ee = new EE();
+
+ee.on('\\nick', function(client, string){
+  client.nickname = string.trim();
+});
+
+ee.on('\\dm', function(client, string){
+  let nickname = string.split(' ').shift().trim();
+  let message = string.split(' ').slice(1).join(' ').trim();
+
+  pool.forEach( c => {
+    if (c.nickname === nickname){
+      c.socket.write(`${client.nickname}: ${message}\n`);
+    }
+  });
+});
+
+ee.on('\\close', function(client, string) {
```

Comment:
I assume this was your intentional implementation for the close event in the assignment. It's meant to be an event that's triggered when a socket closes, not a wack command for the users to use.